pt/utils.c

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright (c) 2022 Intel Corporation.
+
+#include "utils.h"
+
+#define BIT(nr)                 (1UL << (nr))
+#define RTIT_CTL_DISRETC        BIT(11)
+#define RTIT_CTL_TSC_EN         BIT(10)
+#define PT_CTL_DISRETC RTIT_CTL_DISRETC
+#ifndef PT_CTL_TSC_EN
+#define PT_CTL_TSC_EN RTIT_CTL_TSC_EN
+#define PT_CTL_DISRETC RTIT_CTL_DISRETC
+#endif
+
+#define PT_PMU_DIR "/sys/devices/intel_pt/"
+
+int seekPck_w_lib(enum pt_packet_type pt_type, uint64_t *bufEv, long bufsize)
+{
+	struct pt_packet_decoder *decoder;
+	struct pt_config config;
+	uint64_t offset;
+	int r_val = -1;
+	int errcode;
+	struct pt_packet packet;
+	memset(&config, 0, sizeof(config));
+	pt_config_init(&config);
+	config.begin = bufEv;
+	config.end = bufEv + bufsize;
+
+	decoder = pt_pkt_alloc_decoder(&config);
+	if (!decoder) {
+		printf("pt_pkt_alloc_decoder is failed!\n");
+		r_val = -1;
+		return r_val;
+	}	else {
+		errcode = pt_pkt_sync_set(decoder, 0ull);
+		if (errcode < 0) {
+			printf("sync error", 0ull, errcode);
+			r_val = -1;
+			return r_val;
+		}
+	}
+
+	for (;;) {
+		printf("offset = %d\n", offset);
+		errcode = pt_pkt_get_offset(decoder, &offset);
+		if (errcode < 0) {
+			printf("error getting offset %llu %d\r\n", offset, errcode);
+				r_val = -1;
+				return r_val;
+		}
+		errcode = pt_pkt_next(decoder, &packet, sizeof(packet));
+		if (errcode < 0) {
+			if (errcode == -pte_eos)
+				break;
+		}
+		printf("packet.type = %d, pt_type=%d!\n", packet.type, pt_type);
+		if (packet.type == pt_type) {
+			printf("packet.type = %d, pt_type=%d!\n", packet.type, pt_type);
+			printf("package is found!\n");
+			r_val = 0;
+			break;
+		}
+	}
+	return r_val;
+}
+
+/*
+ * Intel_pt pmu
+ */
+int PT_pmu_type(void)
+{
+	FILE *f = fopen(PT_PMU_DIR "type", "r");
+	int type = -1;
+
+	if (f) {
+		if (fscanf(f, "%d", &type) != 1)
+			type = -1;
+		fclose(f);
+	}
+
+	if (type == -1)
+		printf(f ? "PT pmu has a wtf type\n"
+			: "PT pmu not found\n");
+
+	return type;
+}
+/*
+ * initialization event attribut
+ */
+void iniEvtAttr(struct perf_event_attr *attr)
```

Comment:
Do not use Camel-Case naming. All the function reference to this function should be updated.